### PR TITLE
[CDS-43546]: adding linking batch clusters

### PR DESCRIPTION
**Pull request by RajBaviskar**
- [CDS-43546]: adding linking batch clusters

[CDS-43546]: https://harness.atlassian.net/browse/CDS-43546?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helpers/tags.go [hunk 1]
```diff
@@ -25,6 +25,20 @@ func FlattenTags(tags map[string]string) []string {
 	return result
 }
 
+func ExpandClusters(clusterBasicDTO []interface{}) []nextgen.ClusterBasicDto {
```
Comment (rathodmeetsatish) [inline helpers/tags.go:28]: please move this method to cluster resource

File: internal/service/platform/cluster/data_source_cluster.go [hunk 2]
```diff
@@ -38,6 +38,11 @@ func DataSourceCluster() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"scope": {
+				Description: "scope at which the cluster exists in harness gitops",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
```
Comment (rathodmeetsatish) [inline internal/service/platform/cluster/data_source_cluster.go:45]: if this field is not required for get api and we get it in response then it should be computed true

File: helpers/tags.go (hunk 1 above)
Comment (RajBaviskar) [reply helpers/tags.go:28]: Done.

File: internal/service/platform/cluster/data_source_cluster.go (hunk 2 above)
Comment (RajBaviskar) [reply internal/service/platform/cluster/data_source_cluster.go:45]: changed.

**Review by rijajoo — APPROVED** [review]

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**